src/app/loadout/mod-utils.ts

```diff
@@ -299,6 +321,112 @@ export function getCheapestModAssignments(
   return { itemModAssignments: mergedResults, unassignedMods };
 }
 
+/**
+ * For a given item, and mods that need attaching, this creates an ordered list of plugging actions,
+ * to prefer plugging desired mod X into a slot that already has X,
+ * and prefer freeing up armor energy before comsuming it.
+ *
+ * Basically
+ * - It will find all the relevant sockets
+ * - Assign mods to each socket or assign the default plug if there are no mods for the socket
+ * - Then order the results so that we never go over the energy limit if we were to assign them one
+ *   by one.
+ *
+ * THIS ASSUMES THE SUPPLIED ASSIGNMENTS ARE POSSIBLE,
+ * on this item, with its specific mod slots, and will throw if they are not.
+ * this doesn't account for total armor energy, just orders the swaps to avoid overusing energy points.
+ */
+function createOrderedAssignmentResults(
+  defs: D2ManifestDefinitions | undefined,
+  item: DimItem,
+  bucketSpecificAssignments: PluggableInventoryItemDefinition[],
+  bucketIndependentAssignments: PluggableInventoryItemDefinition[]
+) {
+  const pluggingActions: {
+    socketIndex: number;
+    mod?: PluggableInventoryItemDefinition;
+    // This will be negative if we are recovering used energy back by swapping in a cheaper mod
+    energyChange: number;
+  }[] = [];
+
+  const modsToInsert = [...bucketIndependentAssignments, ...bucketSpecificAssignments];
```

Comment:
@nev-r If we want to leave items alone that aren't getting mods assigned to it we want to do a return directly after this if `modsToInsert.length` is 0.

Reply:
I believe this is all that is needed.

```suggestion
  const modsToInsert = [...bucketIndependentAssignments, ...bucketSpecificAssignments];
  if (!modsToInsert.length) {
    return pluggingActions;
  }
```